build.gradle

```diff
@@ -20,6 +20,8 @@ dependencies {
     compile 'com.fasterxml.jackson.core:jackson-core:2.6.4'
     compile 'org.eclipse.paho:org.eclipse.paho.client.mqttv3:1.2.0'
     compile 'org.slf4j:slf4j-api:1.7.13'
+    compile group: 'org.apache.logging.log4j', name: 'log4j-api', version: '2.11.1'
+    compile group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.11.1'
```

Comment:
A library should never depend directly on a logging framework implementation if possible, so we do not have code that requires classes in log4j-core, we should only use log4j-api.

Reply:
Since we use a custom log4j log appender to push on cloud.iO I don't have the choice to use log4j core I think :/